### DISCUSSION TITLE
Looking glass LSP config

### DISCUSSION THREAD

**Discussion by davidbgonz** [nvim-neorg/neorg#811] [General]
Hello. I was wondering if anyone got LSPs to properly work on magnified code blocks? I'm using bashls configured via lspconfig as a test. The filetype is correctly being determined and the LSP is being activated, but for the life of me I cannot figure out how to get the LSP to attach to the magnified code block buffer.

**Comment by ghost** [comment]
Anyone?

**Comment by refaelsh** [comment]
+1

**Comment by max397574** [comment]
most language servers have a single file mode which needs to be set
then it should work with lspconfig in most cases (perhaps try `:LspStart <tab>` to start the server)
otherwise you should do the whole lsp startup manually ig

I never rly made the effort to set this up though

**Comment by tamton-aquib** [comment]
Theres this plugin: https://github.com/jmbuhr/otter.nvim
But i dont know if it works for neorg. But the concept is same and could probably be ported easily i think.

**Comment by refaelsh** [comment]
> perhaps try :LspStart <tab> to start the server

Tried it, it did not work.

> you should do the whole lsp startup manually ig

This is the same as `LspStart` which did not work.
What does "ig" means?

**Comment by refaelsh** [comment]
> I never rly made the effort to set this up though

The "looking glass" feature is the last missing puzzle that prevents me from completely adopting this plugin and its ecosystem.

**Comment by refaelsh** [comment]
> most language servers have a single file mode which needs to be set

It's already set to `true` by default for all the supported LSPs in Neovim: https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md.

**Comment by refaelsh** [comment]
@max397574 Maybe I should open a feature request regarding this?
